app/controllers/sessions_controller.rb

```diff
@@ -141,22 +141,31 @@ def create
         end
       end,
       failure: -> do
-        errors = @handler_result.errors
-        lost_user = errors.any? do |error|
-          [:unknown_callback_state, :invalid_omniauth_data].include? error.code
-        end
 
-        if lost_user
-          # Something weird happened to the user's session or omniauth data and we lost their info
+        def log_and_send_to_root(dev_message, user_alert)
```

Comment:
Can you move this method definition to outside other methods? Having a method define another is not ideal.